### ISSUE TITLE
Handling of harvested records with multiple data_access elements

### ISSUE THREAD

**Issue by steingod** [metno/metsis-drupal#19]
@ferrighi and @magnarem I am struggling with how to handle information that is harvested from a number of sources, including CMEMS and NMDC. These providers expose discovery metadata with multiple data_access elements. I.e. the record is not a strictly dataset element but almost a data collection element. In our system we would have used parent/child relations for this, but these providers do not have the concept and this complicates how to present result in the our portal solution, not to speak about how to work with the basket. Currently we have an understanding that there is one data_access per access type (e.g. direct file, OPeNDAP, WMS etc), but these typically expose a list of files with direct download elements. E.g. IMR may have multiple FTP links for different files to download, but no information on the content in each file. And CMEMS expose multiple WMS and OPeNDAP end points for different variables or products in the same discovery metadata record. 

Is it possible for us to handle this somehow and if so how should we address it? Do we need modifications to MMD? Any ideas?

**Comment by magnarem** [comment]
In solr we support multiple urls in the data_access_<type>_url fields, so It is possible to add multiple http,opendap,etc links to those fields. They should also be listed in the search results and metadata details. however it will be one download button for each link in the list.

**Comment by steingod** [comment]
Sounds good. How would that go with visualisation and basket interaction?

**Comment by magnarem** [comment]
Will investigate and find some solution for this.
Do we have some example datasets? Or I will create some test mmd for this.

**Comment by steingod** [comment]
Not from the top of my mind, but as mentioned several records harvested from NMDC, should be in .../storeB/project/fou/fd/project/adc/mdharvest/imr/mmd

**Comment by ferrighi** [comment]
I agree with Magnar, that by design having a multivalue for the same access type is making this already possible. If you do not have more info about that specific URL, we could maybe think to have the "Download dataset" as a expandable widget? showing the different urls or file names. For the basket I would assume that in the first round all urls should go into it.

**Comment by steingod** [comment]
The issue is that it is hard to inform the user on how to behave. An expandable widget could be an option. In the basket I am not user how we can make useful,, except for download of multiple files which would work. Probably next step would be to establish test based on NPI and NMDC records.

**Comment by magnarem** [comment]
It is possible to implement some pop-up download widget if the multi-fields have more than one entry. This could also provide the possibility to add those multiple datasets to the basket.

**Comment by magnarem** [comment]
> It is possible to implement some pop-up download widget if the multi-fields have more than one entry. This could also provide the possibility to add those multiple datasets to the basket.

The simple_search implemetation have a pop-up download widget, and have support for multiple urls per data_access element types. This form could be further implemented with checkboxes for adding multiple data links to a basket, or create some download bundle if choosing multiple links
